modules/service/src/main/scala/lucuma/odb/service/GuideEnvironmentService.scala

```diff
@@ -0,0 +1,469 @@
+    }
+  }
+
+  private case class ObservationInfo(
+    id:                 Observation.Id,
+    constraints:        ConstraintSet,
+    posAngleConstraint: PosAngleConstraint,
+    optWavelength:      Option[Wavelength],
+    fpu:                Either[GmosNorthFpu, GmosSouthFpu],
+    explicitBase:       Option[Coordinates]
+  ) {
+    def agsParams: AgsParams                  = AgsParams.GmosAgsParams(fpu.some, PortDisposition.Side)
+    def wavelength: Either[Error, Wavelength] =
+      optWavelength.toRight(Error.GeneralError(s"No wavelength defined for observation $id."))
+    def site: Site = fpu.fold(_ => Site.GN, _ => Site.GS)
+  }
+
+  def instantiate[F[_]: Concurrent](
+    httpClient:            Client[F],
+    itcClient:             ItcClient[F],
+    commitHash:            CommitHash,
+    plannedTimeCalculator: PlannedTimeCalculator.ForInstrumentMode
+  )(using Services[F]): GuideEnvironmentService[F] =
+    new GuideEnvironmentService[F] {
+
+      def getAsterism(pid: Program.Id, oid: Observation.Id)(using
+        NoTransaction[F]
+      ): F[Either[Error, NonEmptyList[Target]]] = {
+        asterismService.getAsterism(pid, oid)
+          .map(l => 
+            NonEmptyList.fromList(
+              l.map(_._2)
+            ).toRight(Error.GeneralError(s"No targets have been defined for observation $oid."))
+          )
+      }
+
+      def getObservationInfo(pid: Program.Id, oid: Observation.Id)(using
+        NoTransaction[F]
+      ): F[Either[Error, ObservationInfo]] = {
+        val af = Statements.getObservationInfo(user, pid, oid)
+        session
+          .prepareR(
+            af.fragment.query(Decoders.obsInfoDecoder)
+          )
+          .use(
+            _.option(af.argument).map(_.toRight(Error.GeneralError(s"Observation $oid not found.")))
+          )
+      }
+
+      def getGaiaQuery(
+        oid:             Observation.Id,
+        obsTime:         Instant,
+        tracking:        ObjectTracking,
+        shapeConstraint: ShapeExpression
+      ): Either[Error, ADQLQuery] = {
+        // We want the query dates to be in discreet units to make them more cachable, but
+        // we always want at least six months in the future for the scheduler. So, we'll use one year 
+        // starting at either January or July 1st.
+        val ldt   = LocalDateTime.ofInstant(obsTime, ZoneOffset.UTC)
+        val firstOfYear = ldt.`with`(ChronoField.DAY_OF_YEAR, 1L).`with`(ChronoField.NANO_OF_DAY, 0)
+        val start =
+          if (ldt.getMonthValue() < Month.JULY.getValue())
+            firstOfYear
+          else firstOfYear.`with`(ChronoField.MONTH_OF_YEAR, Month.JULY.getValue())
+        val end   = start.plus(1, ChronoUnit.YEARS)
+
+        (tracking.at(start.toInstant(ZoneOffset.UTC)), tracking.at(end.toInstant(ZoneOffset.UTC)))
+          .mapN { (a, b) =>
+            // Make a query based on two coordinates of the base of an asterism over a year
```

Comment:
What will this do?  The candidates have to be valid for the entire semester?  I'm not sure we want to include scheduler concerns in this query.  That said, this is probably fine since any guide star that moves out of range over the course of the semester is borderline anyway.

Reply:
This was going to be used for some caching, but that is probably not going to be done - at least not here. So, this will probably be reworked in a future PR.